src/NUnitEngine/nunit.engine.core/Internal/TestAssemblyResolver.cs

```diff
@@ -108,7 +108,8 @@ private static string GetDotNetInstallDirectory()
 
         private static string FindBestVersionDir(string libraryDir, Version targetVersion)
         {
-            string target = targetVersion.ToString();
+            if (targetVersion == null)
```

Comment:
It may be clearer to put the null check in the calling code and never call this method with a null version. Since it's a private method and is only called in one place, I don't think a guard is really needed, but you could add one here in addition to the external check.